default to type 2:
// This implementation of type_info does not assume there is always a unique
// copy of the RTTI for a given type inside a program. For various reasons
// the linker may have failed to merge every copy of a types RTTI
// (For example: -Bsymbolic or llvm.org/PR37398). Under this assumption, two
// type_infos are equal if their addresses are equal or if a deep string
// comparison is equal.
this is mostly for compat with libstdc++, which some projects like kde* rely on:
https://bugs.kde.org/show_bug.cgi?id=479679
if https://github.com/llvm/llvm-project/issues/36746 is fixed it might not be needed
--
--- a/libcxx/include/typeinfo
+++ b/libcxx/include/typeinfo
@@ -179,7 +179,7 @@
 
 // On all other platforms, assume the Itanium C++ ABI and use the Unique implementation.
 #      else
-#        define _LIBCPP_TYPEINFO_COMPARISON_IMPLEMENTATION 1
+#        define _LIBCPP_TYPEINFO_COMPARISON_IMPLEMENTATION 2
 #      endif
 #    endif
 
